### ISSUE TITLE
feat: ProtocolEngineImplementation issues load labware command

### ISSUE THREAD

**Issue by mcous** [Opentrons/opentrons#7390]
## Overview

The `load_labware` method of the newly created `ProtocolEngineImplementation` should execute a `load_labware` command on the Protocol Engine.

The `labware_id` returned by the engine response can be used to construct a `LabwareInterface`, but that functionality is out of scope for this ticket.

This ticket pulls off a chunk of #7325 to reduce its scope and feed some extra clarity into #7332.

## Implementation details

1. Add a new interface to the protocol engine that constructs a LoadLabwareRequest and returns a LoadLabwareResult
    - Eventually, we probably want one of these helper methods for every single command
    - The primary consumer of this interface will be the ProtocolContext
2. Call this method within the `ProtocolEngineImplementation`
3. Continue to raise a `NotImplemented` error, as the correct LabwareInterface has not yet been implemented and will be wired up in #7325  

## Acceptance criteria

- [ ] When combined with #7332, #7325 is unblocked
